src/librustc/mir/repr.rs

```diff
@@ -614,12 +618,60 @@ impl<'tcx> Debug for Lvalue<'tcx> {
 }
 
 ///////////////////////////////////////////////////////////////////////////
+// Scopes
+
+#[derive(Clone, Debug, RustcEncodable, RustcDecodable)]
+pub struct ScopeDataVec {
```

Comment:
I think my thinking was that I only wanted `ScopeId` to be usable as an index in *this particular* vector, but `Vec<ScopeData>` is probably enough to ensure that.